packages/graphql-codegen-cli/src/codegen.ts

```diff
@@ -210,8 +215,22 @@ export async function executeCodegen(input: CodegenContext | Types.Config): Prom
                           Object.assign(schemaPointerMap, unnormalizedPtr);
                         }
                       }
-                      outputSchemaAst = await context.loadSchema(schemaPointerMap);
-                      outputSchema = getCachedDocumentNodeFromSchema(outputSchemaAst);
+
+                      const hash = createHmac('sha256', JSON.stringify(schemaPointerMap)).digest('hex');
+
+                      if (schemaMap[hash]) {
+                        outputSchemaAst = schemaMap[hash]['outputSchemaAst'];
+                        outputSchema = schemaMap[hash]['outputSchema'];
+                      } else {
+                        outputSchemaAst = await context.loadSchema(schemaPointerMap);
+                        outputSchema = getCachedDocumentNodeFromSchema(outputSchemaAst);
+
+                        schemaMap[hash] = {
+                          schemaPointerMap: schemaPointerMap,
+                          outputSchemaAst: outputSchemaAst,
+                          outputSchema: outputSchema,
+                        };
```

Comment:
I said it before but we shouldn't cache values, we should share promises. An example, the `context.loadSchema()` part will take let's say 2s and you use the same pointer for 3 generated outputs. Codegen will still load schema 3 times and it will never hit `if (schema[hash])`.

Reply:
because we do the same caching logic for documents and schema, let's write a function with common logic.